module/strategy/WalkInsideBoundedBox/src/WalkInsideBoundedBox.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef MODULE_STRATEGY_WALKINSIDEBOUNDEDBOX_HPP
+#define MODULE_STRATEGY_WALKINSIDEBOUNDEDBOX_HPP
+
+#include <Eigen/Core>
+#include <nuclear>
+
+#include "extension/Behaviour.hpp"
+
+namespace module::strategy {
+
+    class WalkInsideBoundedBox : public ::extension::behaviour::BehaviourReactor {
+    private:
+        /// @brief Stores configuration values
+        struct Config {
+
+            /// @brief Bounded box of region to defend the ball - x min, x max, y max, y min
+            Eigen::Vector4f defending_region = Eigen::Vector4f::Zero();
```

Comment:
I don't understand why you are using a vec4, it seems you are only using it to store four values rather than an actual four dimensional vector.

Reply:
This was Toms idea

Reply:
Could split it up into four variables. It would make code more readable. Also change all the types to `double`

Reply:
Could also make sense as an array of vec2s defining the corners of the box

Reply:
That'd require some fancy stuff to check if inside the shape the four points create, having the four limits makes it much simpler